### PR TITLE
feat: add `array_duplicate_by()` helper to get duplicate array by column

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Add `array_duplicate_by` function to get duplicate array by column.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
I wonder about the overall usefulness of this helper method... I'm not entirely convinced, but if others see potential in it, I won't be opposed.

**Comment by kenjis** [comment]
I wonder too. Can you show use cases?

File: tests/system/Helpers/ArrayHelperTest.php [hunk 1]
```diff
@@ -1287,4 +1287,79 @@ public static function provideArrayGroupByExcludeEmpty(): iterable
...
+                2 => [
+                    'name' => 'Fred Flinstone',
+                ],
+                3 => [
+                    'age' => 30,
+                ],
+            ],
+        ];
+    }
 }
```
Comment (kenjis) [inline tests/system/Helpers/ArrayHelperTest.php:1365]: Do you mean like this?

```suggestion
    public static function provideArrayDuplicate(): iterable
    {
        yield from [
            'single' => [
                'name', // Single
                [
                    [
                        'name' => 'Fred Flinstone',
                        'age'  => 20,
                    ],
                    [
                        'name' => 'Brad Pierce',
                        'age'  => 30,
                    ],
                    [
                        'name' => 'Fred Flinstone',
                        'age'  => 70,
                    ],
                    [
                        'name' => 'Michelle Stone',
                        'age'  => 30,
                    ],
                    [
                        'name' => 'Michael Bram',
                        'age'  => 40,
                    ],
                ],
                [
                    2 => [
                        'name' => 'Fred Flinstone',
                    ],
                ],
            ],
            'multiple' => [
                ['name', 'age'], // Multiple
                [
                    [
                        'name' => 'Fred Flinstone',
                        'age'  => 20,
                    ],
                    [
                        'name' => 'Brad Pierce',
                        'age'  => 30,
                    ],
                    [
                        'name' => 'Fred Flinstone',
                        'age'  => 70,
                    ],
                    [
                        'name' => 'Michelle Stone',
                        'age'  => 30,
                    ],
                    [
                        'name' => 'Michael Bram',
                        'age'  => 40,
                    ],
                ],
                [
                    2 => [
                        'name' => 'Fred Flinstone',
                    ],
                    3 => [
                        'age' => 30,
                    ],
                ],
            ],
        ];
    }
```

File: system/Helpers/array_helper.php [hunk 2]
```diff
@@ -15,6 +15,21 @@
...
+     *
+     * @param array|string $column Unique Column
+     * @param array        $data   Array Data
+     *
+     * @return array
+     */
+    function array_duplicate_by($column, array $data = [])
+    {
+        return ArrayHelper::arrayDuplicatesBy($column, $data);
+    }
```
Comment (kenjis) [inline system/Helpers/array_helper.php:30]: I would use names like the following for consistency with other array helpers.

```suggestion
    /**
     * Returns duplicate elements from an array by key(s).
     *
     * @param array|string $key   Key(s) to check
     * @param array        $array Array Data
     *
     * @return array
     */
    function array_duplicate_by($key, array $array)
    {
        return ArrayHelper::duplicatesBy($key, $array);
    }
```

**Comment by ddevsr** [comment]
@michalsn @kenjis I have real cases in my project

Used for upload batch by excel, I made this so that when uploading `hospital` or `member` master data it immediately rejects duplication without needing to check it from the database. As we all know, to save to the database, doing it by looping and checking in it will take a lot of time.

**Comment by michalsn** [comment]
@ddevsr I do not doubt that you have a use case for this, but I'm not sure if this is such a common problem to add it to the framework.

File: tests/system/Helpers/ArrayHelperTest.php (hunk 1 above)
Comment (ddevsr) [reply tests/system/Helpers/ArrayHelperTest.php]: Done

File: system/Helpers/array_helper.php (hunk 2 above)
Comment (ddevsr) [reply system/Helpers/array_helper.php]: Done

**Comment by ddevsr** [comment]
@kenjis @michalsn I follow decision core team

**Comment by kenjis** [comment]
I posted in the forum.
https://forum.codeigniter.com/showthread.php?tid=89308

**Comment by kenjis** [comment]
There are more people who disagree.

We would encourage people to use libraries over adding functions to the framework core - especially Helpers since they aren’t easily extensible, replaceable, or portable.

**Pull request closed without merging**